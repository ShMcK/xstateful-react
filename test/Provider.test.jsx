import React from 'react';
import renderer from 'react-test-renderer';
import { Machine } from 'xstate';
import { createStatefulMachine } from '@avaragado/xstateful';

import { createReactMachine } from '..';

describe('Provider', () => {
    const machine = Machine({
        key: 'test',
        initial: 'first',
        states: {
            first: {},
        },
    });

    test('ignores value prop', () => {
        const xsf = createStatefulMachine({ machine });
        xsf.init();

        const { Provider, Consumer } = createReactMachine(xsf);

        const component = renderer.create(
            <Provider value="should-ignore-this">
                <Consumer>{value => JSON.stringify(value)}</Consumer>
            </Provider>,
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
