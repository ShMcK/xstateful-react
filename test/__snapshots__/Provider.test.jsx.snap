// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Provider ignores value prop 1`] = `"{\\"state\\":{\\"value\\":\\"first\\",\\"actions\\":[],\\"activities\\":{},\\"data\\":{},\\"events\\":[]},\\"extstate\\":{}}"`;
